# Descripción de los datos

In [16]:
# importa las librerias requeridas
import pandas as pd
import os


In [23]:
# leer el archivo music_proyect_en.csv de la carpeta datasets y guardarlos en una variable
music_df = pd.read_csv('../datasets/raw/music_project_en.csv') 

In [24]:
# mostrar las primeras filas
print(music_df.head(10))

     userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5

In [25]:
music_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB


In [27]:
print('Valores duplicados: ', music_df.duplicated().sum())

Valores duplicados:  3826


Contiene siete columnas. Almacenan los mismos tipos de datos: object.

Según la documentación:

' userID': identificador del usuario o la usuaria;
'Track': título de la canción;
'artist': nombre del artista;
'genre': género de la pista;
'City': ciudad del usuario o la usuaria;
'time': la hora exacta en la que se reprodujo la canción;
'Day': día de la semana.


Podemos ver tres problemas con el estilo en los encabezados de la tabla:

Algunos encabezados están en mayúsculas, otros en minúsculas.
Hay espacios en algunos encabezados.
Encanezados con varias palabras por lo que es necesario usar snake_case.

Observaciones: 

1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas? 

Todos los datos son de tipo 'object' y a traves de la documentación y nombre de las columnas podemos determinar que representa cada columna

2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información? 

Tendriamos que determinar cuales son las ciudades en las que esta contemplado el estudio para poder responder esta pregunta, ya que si son para todo el mundo podriamos decir que los datos no son suficientes, sin embargo podemos delimitar nuestro estudio determinando las ciudades o la región, tambien estamos determinando que solo existe un solo conjunto de datos y no sabemos de donde provienen por lo que podriamos solicitar otras fuentes de datos para poder complementarlos.

3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos? 

En las columnas 'Track','artist' y 'genre' encontramos valores ausentes, y existen 3826 valores que son duplicados por lo que se deberian eliminar.